### ISSUE TITLE
Clustering models by weights

### ISSUE THREAD

**Issue by charlesdgburns** [charlesdgburns/NM_TinyRNN#31]
# Introduction

We are interested in understanding whether the variability in mechanisms can be identified at the level of weights. One idea is to somehow cluster the weights within a model architecture.

However, if we are interested in variability of computations rather than implementation of a common computation, we need to account for symmetries.

Due to the linear -> softmax() decoder, the hidden states may map to logits with permuted hidden units or flipped units.
In order to align the hidden states across models, then
